tests/data/test_converter_public_trades.py

```diff
@@ -0,0 +1,288 @@
+                                   'bid_amount', 'ask_amount',
+                                   'total_volume', 'total_trades']
+    assert 23 == len(df)
+    assert df.index[0] < df.index[1] < df.index[2]
+    assert df.index[0] + BIN_SIZE_SCALE == df.index[1]
+    assert (trades['price'].min() -
+            BIN_SIZE_SCALE) < df.index[0] < trades['price'].max()
+    assert (df.index[0] + BIN_SIZE_SCALE) >= df.index[1]
+    assert (trades['price'].max() -
+            BIN_SIZE_SCALE) < df.index[-1] < trades['price'].max()
+
+    assert 32 == df['bid'].iat[0]  # bid
+    assert 197.512 == df['bid_amount'].iat[0]  # bid
+    assert 88.98 == df['ask_amount'].iat[0]  # ask
+    assert 26 == df['ask'].iat[0]  # ask
+    assert -108.53200000000001 == df['delta'].iat[0]  # delta
+    assert 3 == df['bid'].iat[-1]  # bid
+    assert 50.659 == df['bid_amount'].iat[-1]  # bid
+    assert 108.21 == df['ask_amount'].iat[-1]  # ask
+    assert 44 == df['ask'].iat[-1]  # ask
+    assert 57.551 == df['delta'].iat[-1]  # delta
+
+    BIN_SIZE_SCALE = 1
+    trades = public_trades_to_dataframe(public_trades_list, 'doesntmatter')
+    df = trades_to_volumeprofile_with_total_delta_bid_ask(
+        trades, scale=BIN_SIZE_SCALE)
+    assert 2 == len(df)
+    assert df.index[0] < df.index[1]
+    assert (trades['price'].min() -
+            BIN_SIZE_SCALE) < df.index[0] < trades['price'].max()
+    assert (df.index[0] + BIN_SIZE_SCALE) >= df.index[1]
+    assert (trades['price'].max() -
+            BIN_SIZE_SCALE) < df.index[-1] < trades['price'].max()
+
+    assert 1667.0 == df.index[-1]
+
+    # bid assert 763.7 == df['ask'].iat[0]  # ask
+    assert 710.98 == df['bid_amount'].iat[0]
+    assert 111 == df['bid'].iat[0]
+    assert 52.71999999999997 == df['delta'].iat[0]  # delta
+    # assert 50.659 == df['bid'].iat[-1]  # bid
+    # assert 108.21 == df['ask'].iat[-1]  # ask
+    # assert 57.551 == df['delta'].iat[-1]  # delta
+    #
+
+# bidask
+
+
+def do_plot(pair, data, trades, plot_config=None):
+    import plotly.offline as pyo
+
+    from freqtrade.plot.plotting import generate_candlestick_graph
+
+    # Filter trades to one pair
+    trades_red = trades  # .loc[trades['pair'] == pair].copy()
+    # Limit graph period to your BT timerange
+    data_red = data  # data['2021-04-01':'2021-04-20']
+
+    # plotconf = strategy.plot_config
+    plotconf = plot_config
+
+    # Generate candlestick graph
+    graph = generate_candlestick_graph(pair=pair,
+                                       data=data_red,
+                                       trades=trades_red,
+                                       plot_config=plotconf
+                                       )
+    pyo.plot(graph, output_type="file", show_link=False,
+             filename="tests/data/test_converter_public_trades.html")
+
+
+# need to be at last to see if some test changed the testdata
+# always need to use .copy() to avoid changing the testdata
```

Comment:
Hm is this still the case ? 

we use `--random-order` - so it's surely not a given that this test is running last (not within this file - not globally)... ?